Have a way to not show Blueprint welcome screen

jfaltermeier opened the pull request:
#### What it does
* adds preference whether to always show the welcome page
* check preference from getting started contribution
* add checkbox on welcome screen and sync with preference

Fixes #185

#### How to test
Build and test the setting from preferences and from welcome page.
Test that settings are synced between welcome page and preferences. 

Please note that when the checkbox get unchecked and the welcome page was not closed, it will get restored just like any other view. So in order to test this, the welcome page needs to be closed between reloads. 

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

marcdumais-work commented:
@jfaltermeier you could add `Fixes #185` in your commit message, to link the PR to the related issue. 

see: https://github.com/eclipse-theia/theia-blueprint/issues/185

marcdumais-work commented:
I do not seem  to be able to toggle the new preference from the welcome page (It works from the Preferences UI):

![Peek 2021-12-21 09-47](https://user-images.githubusercontent.com/25749063/146949111-cf18f826-de28-4aac-b693-b0b4e2a70895.gif)

jfaltermeier commented:
It seems that this is not reproducible always. 
![Peek 2021-12-22 08-47](https://user-images.githubusercontent.com/5889696/147055404-6a0e8a69-4ebc-4816-b08f-7e8ae8cf14bd.gif)
However after restarting the application a few times I will sometimes also get into this state. I will investigate further in the new year

jfaltermeier commented:
I've changed the way of handling changes by inspecting the events more closely and rely less on the cached checkbox state. Also I've changed the code to use the `updateValue` method of the preference service. 
I could not reproduce the issue anymore. I've added some info logging as well, so in case it appears again we may check if something is triggered twice or contains wrong values.

marcdumais-work commented:
@jfaltermeier I can't reproduce the issue anymore, thanks!

marcdumais-work reviewed (approved):
LGTM, thanks!

marcdumais-work commented:
Let's merge this!
